internal/dashboard/business/oceanbase/obcluster.go

```diff
@@ -723,3 +734,176 @@ func GetOBClusterStatistic(ctx context.Context) ([]response.OBClusterStatistic,
+	}
+	if obcluster.Status.Status != clusterstatus.Running {
+		return nil, errors.Errorf("OBCluster status is invalid %s", obcluster.Status.Status)
+	}
+
+	// Create OBClusterOperation for restarting observers
+	operation := &v1alpha1.OBClusterOperation{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "restart-observers-",
+			Namespace:    nn.Namespace,
+		},
+		Spec: v1alpha1.OBClusterOperationSpec{
+			Type:      constants.ClusterOpTypeRestartOBServers,
+			OBCluster: nn.Name,
+			RestartOBServers: &v1alpha1.RestartOBServersConfig{
+				OBServers: param.OBServers,
+				OBZones:   param.OBZones,
+				All:       param.All,
+			},
+		},
+	}
+
+	_, err = clients.CreateOBClusterOperation(ctx, operation)
+	if err != nil {
+		return nil, oberr.NewInternal(err.Error())
+	}
+
+	return buildOBClusterResponse(ctx, obcluster)
+}
+
+func DeleteOBServers(ctx context.Context, nn *param.K8sObjectIdentity, param *param.DeleteOBServersParam) (*response.OBCluster, error) {
+	obcluster, err := clients.GetOBCluster(ctx, nn.Namespace, nn.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Get obcluster %s %s", nn.Namespace, nn.Name)
+	}
+	if obcluster.Status.Status != clusterstatus.Running {
+		return nil, errors.Errorf("OBCluster status is invalid %s", obcluster.Status.Status)
+	}
+
+	// Create OBClusterOperation for deleting observers
+	operation := &v1alpha1.OBClusterOperation{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "delete-observers-",
+			Namespace:    nn.Namespace,
+		},
+		Spec: v1alpha1.OBClusterOperationSpec{
+			Type:      constants.ClusterOpTypeDeleteOBServers,
+			OBCluster: nn.Name,
+			DeleteOBServers: &v1alpha1.DeleteOBServersConfig{
+				OBServers: param.OBServers,
+			},
+		},
+	}
+
+	_, err = clients.CreateOBClusterOperation(ctx, operation)
+	if err != nil {
+		return nil, oberr.NewInternal(err.Error())
+	}
+
+	return buildOBClusterResponse(ctx, obcluster)
+}
+
+func ListOBClusterParameters(ctx context.Context, nn *param.K8sObjectIdentity) ([]clustermodel.ParameterItem, error) {
+	obcluster, err := clients.GetOBCluster(ctx, nn.Namespace, nn.Name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Get obcluster %s %s", nn.Namespace, nn.Name)
+	}
+	observerList := v1alpha1.OBServerList{}
+	err = clients.ServerClient.List(ctx, nn.Namespace, &observerList, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", oceanbaseconst.LabelRefOBCluster, nn.Name),
+	})
+	if err != nil {
+		logrus.WithError(err).Error("Failed to list observers")
+		return nil, errors.Wrap(err, "List observers")
+	}
+	conn, err := utils.GetOBConnection(ctx, obcluster, "root", "sys", obcluster.Spec.UserSecrets.Root)
+	if err != nil {
+		logrus.Info("Failed to get OceanBase database connection")
+		return nil, errors.Wrap(err, "Get OceanBase database connection")
+	}
+	parameterItems := make([]clustermodel.ParameterItem, 0)
+	err = conn.QueryList(ctx, &parameterItems, "SHOW PARAMETERS")
```

Comment:
You can use oceanbase-sdk here, that package is designed as a fundamental package and it's ok to use it in oceanbase-dashboard

Reply:
OK. Actually `QueryList` is used in the implementation of API for Listing OBProxy parameter. I copied that implementation here.

Reply:
I have changed to use `conn.ListParametersWithTenantID`